src/frontends/pytorch/src/op/atan2.cpp

```diff
@@ -0,0 +1,100 @@
+    auto const_pi = v0::Constant::create(element::f32, Shape{}, {std::atan(1.0) * 4});
+    // Same input type
+    auto x = context.mark_node(std::make_shared<v1::ConvertLike>(x, const_pi));
+    auto is_x_negative = context.mark_node(std::make_shared<v1::Less>(x, const_zero));
+    auto y = context.mark_node(std::make_shared<v1::ConvertLike>(y, const_zero));
+    auto y_non_negative = context.mark_node(std::make_shared<v1::GreaterEqual>(y, const_zero));
+    auto cond1 = context.mark_node(std::make_shared<v1::LogicalAnd>(is_x_negative, y_non_negative));
+    auto atan = context.mark_node(std::make_shared<v1::ConvertLike>(atan, const_pi));
+    auto atan_y_x_plus_pi = context.mark_node(std::make_shared<v1::Add>(atan, const_pi));
+    result = context.mark_node(std::make_shared<v1::Select>(cond1, atan_y_x_plus_pi, result));
+
+    // handle the third condition : x<0 && y<0
+    auto y = context.mark_node(std::make_shared<v1::ConvertLike>(y, const_zero));
+    auto is_y_negative = context.mark_node(std::make_shared<v1::Less>(y, const_zero));
+    auto cond2 = context.mark_node(std::make_shared<v1::LogicalAnd>(is_x_negative, is_y_negative));
+    auto y = context.mark_node(std::make_shared<v1::ConvertLike>(atan, const_pi));
+    auto atan_y_x_minus_pi = context.mark_node(std::make_shared<v1::Subtract>(atan, const_pi));
+    result = context.mark_node(std::make_shared<v1::Select>(cond2, atan_y_x_minus_pi, result));
+
+    // handle the fourth condition : x=0 && y>0
+    auto x = context.mark_node(std::make_shared<v1::ConvertLike>(x, const_zero));
+    auto is_x_zero = context.mark_node(std::make_shared<v1::Equal>(x, const_zero));
+    auto y = context.mark_node(std::make_shared<v1::ConvertLike>(y, const_zero));
+    auto is_y_positive = context.mark_node(std::make_shared<v1::Greater>(y, const_zero));
+    auto cond3 = context.mark_node(std::make_shared<v1::LogicalAnd>(is_x_zero, is_y_positive));
+    auto const_two = v0::Constant::create(element::f32, Shape{}, {2});
+    auto const_pi = context.mark_node(std::make_shared<v1::ConvertLike>(const_pi, const_two));
+    auto pi_div_two = context.mark_node(std::make_shared<v1::Divide>(const_pi, const_two));
+    result = context.mark_node(std::make_shared<v1::Select>(cond3, pi_div_two, result));
+
+    // handle the fifth condition : x=0 && y<0
+    auto cond4 = context.mark_node(std::make_shared<v1::LogicalAnd>(is_x_zero, is_y_negative));
+    auto const_minus_two = v0::Constant::create(element::f32, Shape{}, {-2});
+    auto const_minus_two = context.mark_node(std::make_shared<v1::ConvertLike>(const_minus_two, const_pi));
+    auto pi_div_minus_two = context.mark_node(std::make_shared<v1::Divide>(const_pi, const_minus_two));
+    result = context.mark_node(std::make_shared<v1::Select>(cond4, pi_div_two, result));
+
+    // check whether out tensor is given
+    if(!context.input_is_none(2) && context.get_input_size() == 3) {
+        context.mutate_input(2, result);            
+    }
```

Comment:
You used `optional_out` decorator in op_table, so you do not need to call `mutate_input` here.